arch/aarch64/mcount-support.c

```diff
@@ -135,7 +136,12 @@ void mcount_get_stack_arg(struct mcount_arg_context *ctx,
 	if (offset < 1 || offset > 100)
 		pr_dbg("invalid stack offset: %d\n", offset);
 
-	memcpy(ctx->val.v, ctx->stack_base + offset, spec->size);
+	addr += offset;
+
+	if (check_mem_region(ctx, (unsigned long)addr))
+		memcpy(ctx->val.v, addr, spec->size);
+	else
+		pr_dbg("stack address is not allowed: %p\n", addr);
```

Comment:
Could you please set `ctx->val.v` to zero for the error cases?  Maybe it'd be better doing it at the beginning of the function unconditionally for possible changes...